test/model/internal_link_test.dart

```diff
@@ -0,0 +1,394 @@
+
+import 'package:checks/checks.dart';
+import 'package:test/scaffolding.dart';
+import 'package:zulip/api/model/model.dart';
+import 'package:zulip/model/internal_link.dart';
+import 'package:zulip/model/narrow.dart';
+import 'package:zulip/model/store.dart';
+
+import '../example_data.dart' as eg;
+import 'test_store.dart';
+
+PerAccountStore setupStore({
+  required Uri realmUrl,
+  List<ZulipStream>? streams,
+  List<User>? users,
+}) {
+  final account = eg.selfAccount.copyWith(realmUrl: realmUrl);
+  final store = eg.store(account: account);
+  if (streams != null) {
+    store.addStreams(streams);
+  }
+  store.addUser(eg.selfUser);
+  if (users != null) {
+    store.addUsers(users);
+  }
+  return store;
+}
+
+void main() {
+  final realmUrl = Uri.parse('https://example.com/');
+
+  checkExpectedNarrows(List<(String, Narrow?)> testCases, {
+    List<ZulipStream>? streams,
+    PerAccountStore? store,
+    List<User>? users,
+  }) {
+    assert((streams != null || users != null) ^ (store != null));
+    store ??= setupStore(realmUrl: realmUrl, streams: streams, users: users);
+    for (final testCase in testCases) {
+      final String urlString = testCase.$1;
+      final Uri url = realmUrl.resolve(urlString);
+      final Narrow? expected = testCase.$2;
+      test(urlString, () {
+        check(parseInternalLink(url, store!)).equals(expected);
```

Comment:
This test helper is violating the contract prescribed in the function's docs:
```dart
/// A [Narrow] from a given URL, on `store`'s realm.
///
/// `url` must already be passed through [tryResolveOnRealmUrl].
```

We should not do that — it undermines the value of the test.